datalad_dataverse/remote.py

```diff
@@ -50,8 +53,31 @@ def transfer_retrieve(self, key, file):
         pass
 
     def remove(self, key):
-        raise
-        pass
+         # connect to dataverse instance
+        api = NativeApi(base_url=self.annex.getconfig('url'),
+                        api_token=os.environ["DATAVERSE_API_TOKEN"])
+        
+        # get the dataset and a list of all files
+        dataset = api.get_dataset(identifier=self.annex.getconfig('doi'))
+        files_list = dataset.json()['data']['latestVersion']['files']
+
+        file_id = None
+
+        # find the file we want to delete
+        for file in files_list:
+            filename = file['dataFile']['filename']
+            if filename == key:
+                file_id = file['dataFile']['id']
+                break
+        
+        if file_id is None:
+            # todo: What to do if the file is not present?
+            raise
+        
+        # delete the file
+        delete(f'{self.annex.getconfig("url")}/dvn/api/data-deposit/v1.1/swordv2/edit-media/file/4', 
```

Comment:
This is an option, of course. However, note that `pydataverse` provides a `<get|delete|post>_request` via `NativeApi` that includes the authentication, since it's set up in that instance already. Kinda reduces the potential to accidentally do a different (or somehow broken) authentication here.

Reply:
Never mind. I didn't see your comment in the issue yet. :)